### DISCUSSION TITLE
Just needing help with compiling

### DISCUSSION THREAD

**Discussion by Shoomie** [recastnavigation/recastnavigation#676] [Q&A]
Short question, I am new to C++ specifically and I can't really make heads and tails out of how this project should even be compiled.

I think it has something to do with using Cmake to render a VS solution which can then be built, but I can't find any .exe still.

Could i get some instructions on how to get started?
Because the Readme leaves a bit to be desired, unless I'm just blind.

**Comment by feiyangbeyond** [comment]
If you just want to compile and run recastdemo, in windows, you need download SDL library into `RecastDemo/Contrib/SDL` dir. Then you can use cmake to build it.

**Comment by grahamboree** [comment]
For people new to the C++ ecosystem, using the premake build steps might be easier.  The general gist is:

1.  Use premake to generate a VS Solution file
2.  Download the SDL libraries and put them in the right place according to the instructions
3.  Build and run the RecastDemo application through Visual Studio

**Reply by cadop** [reply]
I am having the same problem. It isn't clear from the README link what the 'libraries' are. 

1. You can download the entire repo, and unzip it, which does not have a `lib` path. 
2. You can download the release, which only contains a `.dll`
3. You can build SDL, but the instructions don't say that
4. Some step that is obvious to people that use SDL and/or C++ environments alot know what it means, but it would be helpful to specify what needs to move.  

I first thought it was item (2), but then the vs project is looking for `SDL.h` and can't find it (since it just the dll)

**Comment by cadop** [comment]
So I went mad trying to figure out how to solve this.  There wasn't anything at the bottom of the readme for SDL. 

If you go to the "Releases" on the right side:

Click `Assets`
Download `SDL2-devel-2.30.0-VC.zip` (yours might have 2.xx.x)
That is the thing you need!  The instructions are accurate, but just hard to find exactly. 

![image](https://github.com/recastnavigation/recastnavigation/assets/11399119/22aaebc1-edb2-4692-a865-a1023f29a378)

**Reply by grahamboree** [reply]
Feel free to submit a PR if you feel like the "how to build" instructions could be improved!
